internal/pkg/template/templates/task/cf.yml

```diff
@@ -89,9 +99,46 @@ Resources:
             Action: 'sts:AssumeRole'
       ManagedPolicyArns:
         - !Sub 'arn:${AWS::Partition}:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy'
-
-      {{- if or .SSMParamSecrets .SecretsManagerSecrets}}
       Policies:
+        - !If
+          # Optional IAM permission required by ECS task def env file
+          # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/taskdef-envfiles.html#taskdef-envfiles-iam
+          # Example EnvFileARN: arn:aws:s3:::stackset-demo-infrastruc-pipelinebuiltartifactbuc-11dj7ctf52wyf/manual/1638391936/env
+          - HasEnvFile
+          - PolicyName:
+              !Join [
+                "",
+                [
+                  !Ref TaskName,
+                  GetEnvFilePolicy,
+                ],
+              ]
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+                - Effect: "Allow"
+                  Action:
+                    - "s3:GetObject"
+                  Resource:
+                    - !Ref EnvFileARN
+                - Effect: "Allow"
+                  Action:
+                    - "s3:GetBucketLocation"
+                  Resource:
+                    - !Join
+                      - ""
+                      - - "arn:"
+                        - !Ref AWS::Partition
+                        - ":s3:::"
+                        - !Select [
+                          0,
+                          !Split [
+                            "/",
+                            !Select [ 5, !Split [ ":", !Ref EnvFileARN ] ],
+                          ],
+                        ]
+          - !Ref AWS::NoValue
```

Comment:
Is this to guarantee the correctness when there's no policies under `Policies`?

Reply:
I mostly just copied this but I believe it is needed since we're passing the EnvFileARN as a parameter instead of using a template (so now the Policies will always be written since the EnvFileARN could be changed without the template content itself changes). Thanks for the the review!